src/vocgui/views/word_viewset.py

```diff
@@ -0,0 +1,27 @@
+from rest_framework import viewsets
+from rest_framework.authentication import SessionAuthentication, BasicAuthentication
+from rest_framework.permissions import IsAuthenticated
+from vocgui.models import Document
+from vocgui.serializers import DocumentSerializer
+
+
+class WordViewSet(viewsets.ModelViewSet):
+    """
+    Retrieve the list of all default words/documents or get a single record by appending the id of the requested word.
+    """
+
+    serializer_class = DocumentSerializer
+    http_method_names = ["get"]
+    authentication_classes = [SessionAuthentication, BasicAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        """
+        Get the queryset of words/documents
+
+        :return: The queryset of words
+        :rtype: ~django.db.models.query.QuerySet
+        """
+        if getattr(self, "swagger_fake_view", False):
+            return Document.objects.none()
+        return Document.objects.filter(creator_is_admin=True)
```

Comment:
Maybe we should add some additional filters here? I was thinking of
- at least one confirmed image
- audio file should be available
- at least one related training set that is valid and published
- at least one related discipline that is valid and published

Reply:
Yes, sounds reasonable. So in other words, all documents that the app knows of?

Reply:
Exactly

Reply:
Like this? I'm not sure about the audio filter because in the "document-by-trainingset-id" endpoint it seems that also documents without audio are delivered?

Reply:
document-by-trainingset-id is only used within the admin interface to display an image thumbnail within the training set many to many display. See here https://github.com/digitalfabrik/lunes-cms/blob/develop/src/vocgui/static/js/manytomany_overlay.js

Reply:
Are you sure? The endpoint for the manytomany display seems to be the document-by-document-id (the one that uses `permission_classes = [IsAuthenticated]`), and not the one that retrieves the documents by their training set ids:

https://github.com/digitalfabrik/lunes-cms/blob/c21f2c59de7d414de3bbd810783dbcefa9a2c960/src/vocgui/views/document_viewset.py#L38-L41

Reply:
Sorry, I didn't carefully. Of course I meant document-by-document-id  regarding the many to many overlay

Reply:
Ah I guess in the app we have a fallback whenever an audio file is missing. It uses text-to-speech. So I guess we can neglect it here :-)